base/commands/migration/start_stages_it_test.go

```diff
@@ -0,0 +1,133 @@
+		},
+		{
+			name: "failure",
+			statusMapStateFiles: []string{
+				"testdata/start/migration_success_initial.json",
+				"testdata/start/migration_success_failure.json",
+			},
+			expectedErr: errors.New("Failed migrating IMAP: imap5: some error"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			startMigrationTest(t, tc.expectedErr, tc.statusMapStateFiles)
+		})
+	}
+}
+
+func startMigrationTest(t *testing.T, expectedErr error, statusMapStateFiles []string) {
+	tcx := it.TestContext{T: t}
+	ctx := context.Background()
+	tcx.Tester(func(tcx it.TestContext) {
+		ci := hz.NewClientInternal(tcx.Client)
+		createMapping(ctx, tcx)
+		createMemberLogs(t, ctx, ci)
+		defer removeMembersLogs(ctx, ci)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		var execErr error
+		go tcx.WithReset(func() {
+			defer wg.Done()
+			execErr = tcx.CLC().Execute(ctx, "start", "dmt-config", "--yes")
+		})
+		c := make(chan string, 1)
+		wg.Add(1)
+		go findMigrationID(ctx, tcx, c)
+		mID := <-c
+		wg.Done()
+		wg.Add(1)
+		go migrationRunner(ctx, tcx, mID, &wg, statusMapStateFiles)
+		wg.Wait()
+		if expectedErr == nil {
+			require.Equal(t, nil, execErr)
+		}
+		if expectedErr != nil {
+			require.Contains(t, execErr.Error(), expectedErr.Error())
+		}
+		tcx.WithReset(func() {
+			f := fmt.Sprintf("migration_report_%s.txt", mID)
+			require.Equal(t, true, fileExists(f))
+			Must(os.Remove(f))
+			b := MustValue(os.ReadFile(paths.ResolveLogPath("test")))
+			for _, m := range ci.OrderedMembers() {
+				require.Contains(t, string(b), fmt.Sprintf("[%s_%s] log1", mID, m.UUID.String()))
+				require.Contains(t, string(b), fmt.Sprintf("[%s_%s] log2", mID, m.UUID.String()))
+				require.Contains(t, string(b), fmt.Sprintf("[%s_%s] log3", mID, m.UUID.String()))
+			}
+		})
+	})
+}
+
+func migrationRunner(ctx context.Context, tcx it.TestContext, migrationID string, wg *sync.WaitGroup, statusMapStateFiles []string) {
+	statusMap := MustValue(tcx.Client.GetMap(ctx, migration.StatusMapName))
+	for _, f := range statusMapStateFiles {
+		b := MustValue(os.ReadFile(f))
+		Must(statusMap.Set(ctx, migrationID, serialization.JSON(b)))
+		time.Sleep(2 * time.Second)
+	}
+	wg.Done()
+}
+
+func createMapping(ctx context.Context, tcx it.TestContext) {
+	mSQL := fmt.Sprintf(`CREATE MAPPING IF NOT EXISTS %s TYPE IMap OPTIONS('keyFormat'='varchar', 'valueFormat'='json')`, migration.StatusMapName)
+	MustValue(tcx.Client.SQL().Execute(ctx, mSQL))
+}
+
+func findMigrationID(ctx context.Context, tcx it.TestContext, c chan string) {
+	q := MustValue(tcx.Client.GetQueue(ctx, migration.StartQueueName))
+	var b migration.ConfigBundle
+	for {
+		v := MustValue(q.PollWithTimeout(ctx, time.Second))
+		if v != nil {
+			Must(json.Unmarshal(v.(serialization.JSON), &b))
+			c <- b.MigrationID
+			break
+		}
+	}
+}
+
+func fileExists(filename string) bool {
+	MustValue(os.Getwd())
+	_, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
```

Comment:
Is this function necessary?
There is the `paths.Exists` function.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/398/commits/f50a1de3621a0e85a2f50804eca9223edc4fd15c